myApp/app/components/BuildingPopUp.tsx

```diff
@@ -50,7 +52,18 @@ export const BuildingPopup: React.FC<BuildingPopupProps> = ({ visible, onClose,
               <MaterialIcons name="directions" size={20} color="white" />
               <Text style={styles.buttonText}>Get Directions</Text>
             </TouchableOpacity>
+            <TouchableOpacity style={[styles.button, styles.startPointButton]} 
+            onPress={() => {
+            useAsStartingPoint(building);
```

Comment:
Using the sonarqube extension detected this issue "React Hook "useAsStartingPoint" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function". You need to rename useAsStartingPoint to setAsStartingPoint.
```suggestion
            setAsStartingPoint(building);
```

Reply:
Sounds good. Fixed in latest commit